Stellar-ledger.x

```diff
@@ -407,7 +407,7 @@ struct TransactionMetaV3
                                         // applied if any
     ContractEvent events<>;             // custom events populated by the
                                         // contracts themselves.
-    SCVal returnValue;                  // return value of the invocation.
+    SCVal* returnValue;                 // return value of the host fn invocation (if any)
```

Comment:
Should the Soroban elements be grouped into a struct that as a whole is `None` for other operations? It feels like a data smell that we have multiple fields that are only relevant in one situation.

Reply:
Yeah, that probably would be better. What do you think @sisuresh ?

Reply:
Yeah that'll work well now that we don't hash meta components. Related to this, should we include an `ExtensionPoint` at the top of `TransactionMetaV3` and/or in this new struct?

Reply:
I don't have a strong preference, but since meta itself is an 'extension', do we really need it? Or do you want this to be the final meta version and all the future extensions to happen via extension point? But then if we want to remove something, things become confusing...

Reply:
Only reason I mentioned it is because a V4 will require us to copy everything over. If we want to remove a field, we'll have to do a V4 which I guess would get a little confusing.

Reply:
Ok, added the extension point and updated meta to use soroban-specific struct.